src/preparers.py

```diff
@@ -595,3 +595,48 @@ def prepare_agentinstruct(row):
                 "parent": dataset['id'].split('_')[0] if i == -1 else i,
             })
     return messages
+
+
+def prepare_lumos_planning(row):
+    messages = row["messages"]
+    new_messages = list()
+    for i, message in enumerate(messages):
+        if i == 0:
+            new_messages.append({
+                "from": message["role"],
+                "text": message["content"],
+                "parent": row["dataset"]
+            })
+        else:
+            new_messages.append({
+                "from": message["role"],
+                "text": message["content"],
+                "parent": 0
```

Comment:
Are there multiple turns in these datasets? The parent should be in the index of the message that this is responding to. So if the message history looks like [user, assistant, user, assistant, user, assistant] then the parents will be [<dataset-name>, 0, 1, 2, 3, 4].

Reply:
Fixed in the new update by changing 0 to i-1!